admin/google_search_console/class-gwt-service.php

```diff
@@ -0,0 +1,171 @@
+	 * Returns the client
+	 *
+	 * @return Yoast_Google_Client
+	 */
+	public function get_client() {
+		return $this->client;
+	}
+
+	/**
+	 * Get all sites that are registered in the GWT panel
+	 *
+	 * @return array
+	 */
+	public function get_sites() {
+		$sites = array();
+
+		// Do list sites request.
+		$response = $this->client->do_request( 'sites' );
+
+		if ( $response_json = $this->client->decode_response( $response ) ) {
+			if ( ! empty( $response_json->siteEntry ) ) {
+				foreach ( $response_json->siteEntry as $entry ) {
+					$sites[ str_ireplace( 'sites/', '', (string) $entry->siteUrl ) ] = (string) $entry->siteUrl;
+				}
+			}
+
+			// Sorting the retrieved sites.
+			asort( $sites );
+		}
+
+		return $sites;
+	}
+
+	/**
+	 * Get crawl issues
+	 *
+	 * @return array
+	 */
+	public function get_crawl_issue_counts() {
+
+		// Setup crawl error list.
+		$crawl_error_counts = $this->get_crawl_error_counts( $this->get_profile() );
+
+		$return = array();
+		if ( ! empty( $crawl_error_counts->countPerTypes ) ) {
+			foreach ( $crawl_error_counts->countPerTypes as $category ) {
+				$return[ $category->platform ][ $category->category ] = array(
+					'count'      => $category->entries[0]->count,
+					'last_fetch' => null,
+				);
+			}
+		}
+
+		return $return;
+	}
+
+	/**
+	 * Sending request to mark issue as fixed
+	 *
+	 * @param string $url
+	 * @param string $platform
+	 * @param string $category
+	 *
+	 * @return bool
+	 */
+	public function mark_as_fixed( $url, $platform, $category ) {
+		$profile  = $this->get_profile();
+		$response = $this->client->do_request( 'sites/' .  urlencode( $profile ) .  '/urlCrawlErrorsSamples/' . urlencode( ltrim( $url, '/' ) ) . '?category=' . $category . '&platform=' . $platform . '', 'DELETE' );
+		return ( $response->getResponseHttpCode() === 204 && $response->getResponseBody() === '' );
+	}
+
+	/**
+	 * Get the GWT profile
+	 *
+	 * @return string
+	 */
+	public function get_profile() {
+		// Get option.
+		$option = get_option( self::OPTION_WPSEO_GWT, array( 'profile' => '' ) );
+
+		// Set the profile.
+		$profile = $option['profile'];
+
+		// Backwards compatibility fix - This is the old API endpoint.
+		if ( strpos( $profile, 'https://www.google.com/webmasters/tools/feeds/' ) ) {
+			$profile = str_replace( 'https://www.google.com/webmasters/tools/feeds/', '', $profile );
+		}
+
+		// Return the profile.
+		return trim( $profile, '/' );
+	}
+
+	/**
+	 * Fetching the issues from the GWT API
+	 *
+	 * @param string $platform
+	 * @param string $category
+	 *
+	 * @return mixed
+	 */
+	public function fetch_category_issues( $platform, $category ) {
+		$response = $this->client->do_request(
+			'sites/'. urlencode( $this->get_profile() ) . '/urlCrawlErrorsSamples?category=' . $category . '&platform=' . $platform
+		);
+
+		if ( $issues = $this->client->decode_response( $response ) ) {
+			if ( ! empty ($issues->urlCrawlErrorSample ) ) {
+				return $issues->urlCrawlErrorSample;
+			}
+		}
```

Comment:
This function needs a `return false;` at the end. That way it is explicit you are returning something. PHPStorm should nag you about this already.